### DISCUSSION TITLE
Unable to resolve local dependencies with PNPM Monorepos

### DISCUSSION THREAD

**Discussion by ale-ben** [vercel/community#5225] [Help]
### Summary

I decided to merge all my project related repos in a monorepo.
I used PNPM workspaces and NX.
I reconfigured vercel to work with a subfolder and everything worked fine.

After that I started exporting common functions as separate packages and importing them locally using PNPM workspaces.
Since I have done that, vercel is not able to resolve my workspace packages.
"Include source files outside ..." is already enabled.

Note that running the same command vercel runs ( `cd ../.. && npx nx build backend` ) locally produces a valid build with no errors.

Last deployment is here [teal-owl/Fy2aUZarJoFqx68DLykRSmNrUFq7](https://vercel.com/alessandro-benettons-projects/teal-owl/Fy2aUZarJoFqx68DLykRSmNrUFq7) and the repository is here [https://github.com/ale-ben/Teal-Owl](https://github.com/ale-ben/Teal-Owl).

Any help is really appreciated

### Steps to Reproduce

1. Generate a pnpm workspace
2. Generate a package.json for a library under /libs/... and export a function
3. Generate an app under /apps/...
4. Import in the app's package.json the library's package ( `pnpm add ...` )
5. Try to deploy, Vercel won't be able to resolve the imported library

**Comment by ale-ben** [comment]
Was about to give up but I found the  transpilePackages option in next config and that solved everything up!
[https://nextjs.org/docs/app/api-reference/next-config-js/transpilePackages](https://nextjs.org/docs/app/api-reference/next-config-js/transpilePackages)

**Comment by smogg** [comment]
Any solutions for projects not using Next.js?
